src/tests/performance/ping_pong.rs

```diff
@@ -0,0 +1,196 @@
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │      50 │        150 │          0 │         45 │              3 │          0 │          0 │          0 │          0 │          0 │          10.13 │      19.42 │        39.14 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │      75 │        150 │          0 │         57 │              6 │          0 │          0 │          0 │          0 │         41 │           8.23 │      25.28 │        36.63 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     100 │        150 │          0 │         16 │              2 │          0 │          0 │          0 │          0 │          1 │           7.21 │      29.92 │        36.13 │
+    // └─────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴────────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴──────────────┘
+    //
+    // *NOTE* run with `cargo test --release tests::performance::ping_pong -- --nocapture`
+
+    let synth_counts = vec![1, 10, 15, 20, 30, 50, 75, 100];
+
+    let mut table = RequestsTable::default();
+
+    let target = TempDir::new().expect("Unable to create TempDir");
+    let mut node = Node::builder()
+        .start(target.path(), NodeType::Stateless)
+        .await
+        .unwrap();
+    let node_addr = node.addr();
+
+    for synth_count in synth_counts {
+        // setup metrics recorder
+        let test_metrics = TestMetrics::default();
+        // clear metrics and register metrics
+        metrics::register_histogram!(METRIC_LATENCY);
+
+        let mut synth_handles = Vec::with_capacity(synth_count);
+        let test_start = tokio::time::Instant::now();
+        for _ in 0..synth_count {
+            synth_handles.push(tokio::spawn(simulate_peer(node_addr)));
+        }
+
+        // wait for peers to complete
+        for handle in synth_handles {
+            let _ = handle.await;
+        }
+
+        let time_taken_secs = test_start.elapsed().as_secs_f64();
+
+        let snapshot = test_metrics.take_snapshot();
+        if let Some(latencies) = snapshot.construct_histogram(METRIC_LATENCY) {
+            if latencies.entries() >= 1 {
+                // add stats to table display
+                table.add_row(RequestStats::new(
+                    synth_count as u16,
+                    PINGS,
+                    latencies,
+                    time_taken_secs,
+                ));
+            }
+        }
+    }
+
+    node.stop().unwrap();
+
+    // Display results table
+    println!("{}", table);
```

Comment:
I think `\r\n` should be before `{}` - if there are no errors then table's header is broken.